### PR TITLE
US112499 Submission and Completion component

### PR DESCRIPTION
**Pull request by pureooze**
* Created the `d2l-activity-assignment-editor-submission-and-completion` component
* Moved related parts of the editor into this component
* Will make separate components for each of these in a separate PR

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-import './d2l-assignment-turnitin-editor';
 import 'd2l-inputs/d2l-input-checkbox.js';
 import 'd2l-inputs/d2l-input-checkbox-spacer.js';
 import '@brightspace-ui-labs/accordion/accordion-collapse.js';
-import { bodySmallStyles, heading4Styles, labelStyles } from '@brightspace-ui/core/components/typography/styles.js';
 import { css, html, LitElement } from 'lit-element/lit-element.js';
 import { AssignmentEntity } from 'siren-sdk/src/activities/assignments/AssignmentEntity.js';
 import { EntityMixinLit } from 'siren-sdk/src/mixin/entity-mixin-lit.js';
 import { getLocalizeResources } from '../localization.js';
+import { labelStyles } from '@brightspace-ui/core/components/typography/styles.js';
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js]: This is due to a linter rule `Imports should be sorted alphabetically            sort-imports`

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+					@change="${this._saveSubmissionTypeOnChange}"
+					?disabled="${!this._canEditSubmissionType}">
+						${this._getSubmissionTypeOptions()}
+				</select>
+			</div>
+		`;
+	}
+
+	_renderAssignmentSubmissionTypeSummary() {
+		return html``;
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js]: I want to implement the actual summary stuff in a separate PR, this one was getting a bit big

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
+import '../d2l-activity-availability-dates-editor.js';
+import '../d2l-activity-release-conditions-editor.js';
+import './d2l-assignment-turnitin-editor';
+import './d2l-activity-assignment-editor-submission-and-completion.js';
 import './d2l-activity-assignment-availability-editor.js';
 import './d2l-activity-assignment-type-editor.js';
-import './d2l-assignment-turnitin-editor';
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js]: we are going to conflict so hard hahahahaha

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+import { LocalizeMixin } from '@brightspace-ui/core/mixins/localize-mixin.js';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+import { SaveStatusMixin } from '../save-status-mixin.js';
+import { selectStyles } from '@brightspace-ui/core/components/inputs/input-select-styles.js';
+
+class AssignmentEditorSubmissionAndCompletion extends SaveStatusMixin(RtlMixin(EntityMixinLit(LocalizeMixin(LitElement)))) {
+
+	static get properties() {
+		return {
+			href: { type: String },
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js]: `EntityMixinLit` does these for you I think, and it puts like `reflect` on `href`.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js [hunk 5]
```diff
@@ -1,19 +1,21 @@
+import '../d2l-activity-availability-dates-editor.js';
+import '../d2l-activity-release-conditions-editor.js';
+import './d2l-assignment-turnitin-editor';
+import './d2l-activity-assignment-editor-submission-and-completion.js';
 import './d2l-activity-assignment-availability-editor.js';
 import './d2l-activity-assignment-type-editor.js';
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js]: I think this should be unused now.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
-import './d2l-assignment-turnitin-editor';
 import 'd2l-inputs/d2l-input-checkbox.js';
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js]: Between our two PRs, there are going to be more unused things. We should coordinate that in whoever loses the race to merge. I imagine the linter will complain about the unused ESM imports (but it won't for the weirdo blank imports).

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js [hunk 7]
```diff
@@ -152,15 +89,6 @@ class AssignmentEditorSecondary extends SaveStatusMixin(RtlMixin(EntityMixinLit(
 				.token="${this.token}">
 			</d2l-activity-rubrics-list-container>
 
-			<div id="assignment-release-conditions-container">
```
Comment (kieranderson) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-secondary.js]: 🙌 Thank you

**Pull request merged**